libcaf_core/caf/actor_system.hpp

```diff
@@ -524,7 +534,9 @@ class actor_system {
 
   /// Calls all thread started hooks
   /// @warning must be called by thread which is about to start
-  void thread_started();
+  void thread_started(const thread_type tt=other_thread);
+
+  void set_affinity(const thread_type tt);
```

Comment:
This entire PR essentially boils down to adding this function. Could you please add documentation to it? From browsing through the code, I think this does the following:
- Calling this function affects the current thread only.
 -It picks the next core for that affinity group in round-robin order. This means calling `set_affinity` N+1 times will bind the the thread to the first core again, given group size N. Thus, two threads can share one core but only within the affinity group.
- This does nothing if not configuring the corresponding type-to-cores parameters.

That's my understanding after reading the code, but a user should receive this information upfront. 🙂

Reply:
Yes, you are correct the code does exactly that. I will add some documentation on the code in the next commit.